pvlib/horizon.py

```diff
@@ -0,0 +1,646 @@
+    profile.
+
+    Parameters
+    ----------
+    horizon_azimuths: numeric
+        Azimuth values for points that define the horizon profile. The ith
+        element in this array corresponds to the ith element in horizon_angles.
+
+    horizon_angles: numeric
+        Elevation angle values for points that define the horizon profile. The
+        elevation angle of the horizon is the angle that the horizon makes with
+        the horizontal. It is given in degrees above the horizontal. The ith
+        element in this array corresponds to the ith element in
+        horizon_azimuths.
+
+    surface_tilt : numeric
+        Surface tilt angles in decimal degrees. surface_tilt must be >=0
+        and <=180. The tilt angle is defined as degrees from horizontal
+        (e.g. surface facing up = 0, surface facing horizon = 90)
+
+    surface_azimuth : numeric
+        Surface azimuth angles in decimal degrees. surface_azimuth must
+        be >=0 and <=360. The azimuth convention is defined as degrees
+        east of north (e.g. North = 0, South=180 East = 90, West = 270).
+
+    Returns
+    -------
+    dtf: numeric
+        The diffuse tilt factor that can be multiplied with the diffuse
+        horizontal irradiance (DHI) to get the incident irradiance from
+        the sky that is adjusted for the horizon profile and the tilt of
+        the plane.
+
+    """
+    assert(horizon_azimuths.shape[0] == horizon_angles.shape[0])
+    tilt_rad = np.radians(surface_tilt)
+    plane_az_rad = np.radians(surface_azimuth)
+    a = np.sin(tilt_rad) * np.cos(plane_az_rad)
+    b = np.sin(tilt_rad) * np.sin(plane_az_rad)
+    c = np.cos(tilt_rad)
+
+    # this gets either a float or an array of zeros
+    dtf = np.multiply(0.0, surface_tilt)
+    num_points = horizon_azimuths.shape[0]
+    for i in range(horizon_azimuths.shape[0]):
+        az = np.radians(horizon_azimuths[i])
+        horizon_elev = np.radians(horizon_angles[i])
+        temp = np.radians(collection_plane_elev_angle(surface_tilt,
+                                                      surface_azimuth,
+                                                      horizon_azimuths[i]))
+        elev = np.maximum(horizon_elev, temp)
+
+        first_term = .5 * (a*np.cos(az) + b*np.sin(az)) * \
+                          (np.pi/2 - elev - np.sin(elev) * np.cos(elev))
+        second_term = .5 * c * np.cos(elev)**2
+        dtf += 2 * (first_term + second_term) / num_points
+    return dtf
+
+
+def DNI_horizon_adjustment(horizon_angles, solar_zenith, solar_azimuth):
+    '''
+    Calculates an adjustment to DNI based on a horizon profile. The adjustment
+    is a vector of binary values with the same length as the provided
+    solar position values. Where the sun is below the horizon, the adjustment
+    vector is 0 and it is 1 elsewhere. The horizon profile must be given as a
+    vector with 361 values where the ith value corresponds to the ith degree
```

Comment:
Is the 361st value needed?  I expected 0 - 359, since the last value at 360 should be a duplicate of the first value at 0.